### DISCUSSION TITLE
Percentage of labeled data needed for estimating directly

### DISCUSSION THREAD

**Discussion by jlb52** [moj-analytical-services/splink#1609] [Q&A] [answered]
Hi all,

I'm curious if anyone has tested the efficacy of estimating directly instead of via expectation maximisation when only a small percentage of labels (e.g. 0.1%) is available. I'm dealing with a very messy dataset with inconsistently formatted full names, many nulls, etc., and am having trouble reaching accurate results via estimation maximisation in my de-duplication effort. 

I'm considering using the correct cluster labels we found when reviewing a sample of our originally generated predictions and feeding these in for the parameter estimates, but given the size of the entire dataset, this isn't a large percentage at all, so I'm uncertain of how well it will work.

If anyone has experience with this, I'd very much appreciate hearing about your experience!

Many thanks,
Jackson

**Comment by RobinL** [comment, marked as answer]
Generally where you have labels I would train the m values of the model from them rather than using expectation maximisation.  In most cases, especially if you don't have many labels, I would continue to train u using [estimate_u_using_random_sampling](https://moj-analytical-services.github.io/splink/linker.html?h=estimate_u#splink.linker.Linker.estimate_u_using_random_sampling)

Re your I think there are two main considerations:
1. Whether your labels are representative of the whole dataset or biased.  
2. Whether you have enough labels (in absolute numbers)

On (2) one of the issues is if you don't have many labels, you may have no examples at all of some of the scenarios you are modelling.  e.g. you may not have a singled labelled pair of a first name at jaro <0.88 or something.  To get a decent estimate, you will want at least a few labelled pairs falling into each of your ComparisonLevels

**Reply by jlb52** [reply]
Makes complete sense, and thanks for the response. I'd thought about your first consideration but hadn't thought of the second so that's very helpful and something that I'll verify.

I think I'll give m estimates a try with the labels and will report back on my results.

**Comment by aflaxman** [comment]
I thought this would be fun to investigate with some simulated data from `pseudopeople`.  Here is [a colab notebook with an initial attempt](https://colab.research.google.com/drive/1nuYR2ZSu-s3WOmn3pCXkAX1jIL0R2G3M?usp=sharing). Comments welcome!

**Reply by jlb52** [reply]
Thanks a lot for this! I finally was able to test on my dataset as well. Unable to include full code/data description for confidentiality reasons, but I'm dropping the comparison of match weight estimates below. 

For reference, I have approximately 300,000 rows of data for deduplication and trained on only 1,000 data, all with the same surname (name being one of the features in the dataset). Parameter estimates are very similar save for an "Else" comparison for one of the features (number 5). To me, it's a strong vote of confidence for the reliability and robustness of the EM trained parameters.

![e9745677-0681-440d-8c32-9e55351ed756](https://github.com/moj-analytical-services/splink/assets/144322305/275d4355-5735-431a-8587-00efc16b9bb3)
